the-collab-lab/tcl-34-smart-shopping-list
# Create List with a random token

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

A shopping list consists of a set of items associated with a user’s token. Creating a new list consists of the following sequence:

Generate a new, unique token
Save the token to localStorage
Show the user the list view

## Related Issue

Closes #3

## Acceptance Criteria

@the-collab-lab/shopping-list-utils has been added as a dependency to the project.
For users who do not already have a token or a list, a button or link exists on the home screen that allows them to create a new list
Clicking the button/link generates a new token and saves it to localStorage
Once the token has been created and saved, the user is redirected to the "list" view
For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/74724204/137225231-dce2f466-e770-413a-a75e-51e424377ab3.png)



### After

![image](https://user-images.githubusercontent.com/74724204/137311517-00b1a525-c5c4-4c8f-8f0e-865cef18b076.png)



## Testing Steps / QA Criteria

1.	From your terminal, pull down this branch with `git pull origin mm-yu-create-list`  and check that branch out with `mm-yu-create-list `
2.	Then npm install the newly added dependencies locally and npm start to launch the app.
3. Click the button "create new list", generates a new token and saves it to localStorage. Once the token has been created and saved, the user is redirected to the "list" view
